### PR TITLE
8188073: Add Capstone as backend for hsdis

### PR DESCRIPTION
**Pull request by magicus**
The Capstone library is a simple to use, efficient disassembly library, distributed under the MIT license. 

This PR implements an hsdis backend using Capstone. It has been tested on Linux, macOS and Windows (x64).

The actual C implementation of the backend was done by @JornVernee. I assume there are plenty of room for enhancing this implementation, with options like in the binutils backend. Such improvements can be done later, by teams more familiar with the requirements of hsdis.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8188073](https://bugs.openjdk.java.net/browse/JDK-8188073): Add Capstone as backend for hsdis

### Reviewers
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**)

### Contributors
 * Magnus Ihse Bursie `<ihse@openjdk.org>`
 * Jorn Vernee `<jvernee@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7506/head:pull/7506` \
`$ git checkout pull/7506`

Update a local copy of the PR: \
`$ git checkout pull/7506` \
`$ git pull https://git.openjdk.java.net/jdk pull/7506/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7506`

View PR using the GUI difftool: \
`$ git pr show -t 7506`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7506.diff">https://git.openjdk.java.net/jdk/pull/7506.diff</a>

</details>

### CODE REVIEW

File: make/Hsdis.gmk [hunk 1]
```diff
@@ -129,13 +151,18 @@ TARGETS += build
 INSTALLED_HSDIS_NAME := hsdis-$(OPENJDK_TARGET_CPU_LEGACY_LIB)$(SHARED_LIBRARY_SUFFIX)
 
 INSTALLED_HSDIS := $(INSTALLED_HSDIS_DIR)/$(INSTALLED_HSDIS_NAME)
+INSTALLED_HSDIS_IMAGE := $(IMAGE_HSDIS_DIR)/$(INSTALLED_HSDIS_NAME)
 
 $(INSTALLED_HSDIS): $(BUILD_HSDIS_TARGET)
-	$(call LogWarn, NOTE: The resulting build might not be redistributable. Seek legal advice before distibuting.)
+ifeq ($(HSDIS_BACKEND), binutils)
```
Comment (erikj79) [inline make/Hsdis.gmk:157]: Don't we usually indent conditionals to the recipe level using spaces?

**Review by erikj79 — APPROVED** [review]
One style nit.

File: make/Hsdis.gmk (hunk 1 above)
Comment (magicus) [reply make/Hsdis.gmk:157]: Yes we do. Embarrassing. 😊

**Pull request closed without merging**